bin/document_slug_updater

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env ruby
+
+require File.expand_path("../../config/environment", __FILE__)
+require "./app/lib/specialist_publisher.rb"
+
+def strip_leading_slash(path)
+  path.sub(%r{^/}, "")
+end
+
+slug     = strip_leading_slash(ARGV.fetch(0))
+new_slug = strip_leading_slash(ARGV.fetch(1))
+
+edition = SpecialistDocumentEdition.published.where(slug: slug).last
+raise(ArgumentError, "No published document for #{slug}") unless edition
+
+doc_type = edition.document_type
+document_id = edition.document_id
+services = SpecialistPublisher.document_services(doc_type)
+
+puts "Changing #{slug} to #{new_slug}"
+
+puts "Withdrawing #{slug}"
+services.withdraw(document_id).call
+
+puts "Creating a new draft to republish"
+edition.update_attribute(:slug, new_slug)
+
+# Without reloading the record, the edition would be left in the "archived" state
+edition.reload
+
+puts "Republishing as #{new_slug}"
+services.publish(document_id).call
```

Comment:
We shouldn't be using the `publish` service here. I renamed the old `republish` service to `bulk_publish` and slightly modified the behaviour, so the call documented in the ops manual isn't possible any more.

I'll add a proper `republish` service that we can use here.